redditdownload/remotemd5.py

```diff
@@ -0,0 +1,40 @@
+"""Python MD5 of remote file (URL)."""
+import hashlib
+import optparse
+try:
+    from urllib2 import urlopen
+except ImportError:
+    from urllib.request import urlopen
+
+
+def get_remote_md5_sum(url, max_file_size=100*1024*1024):
+    """get remote md5 sum.
+
+    Args:
+        url (str): remote url.
+        max_file_size(int): the limit for filesize. default to 100mb.
+
+    Returns:
+        str: md5 of the remote url.
+    """
+    remote = urlopen(url)
+    hash = hashlib.md5()
+
+    total_read = 0
+    while True:
+        data = remote.read(4096)
```

Comment:
Reminder: this is file download. Doing that to check whether the file should be downloaded is pointless.

If an md5 check is needed, it can be done after the main download code (probably by reading the file back from the filesystem).